src/App.js

```diff
@@ -16,60 +16,92 @@ import apiEndpoints from "./config";
 import { cleanJSON } from "./utils";
 
 const App = () => {
-  const [columns, setColumns] = useState([]);
+  const searchParams = new URLSearchParams(window.location.search);
+
+  const [path, setPath] = useState(searchParams.get("path"));
+  const [columns, setColumns] = useState([
+    { items: [], selected: "published" },
+    ...(path?.split(".").map((element) => ({ items: [], selected: element })) ||
+      []),
+  ]);
   const [dataView, setDataView] = useState("");
   const [apiEndpoint, setApiEndpoint] = useState(apiEndpoints[0].value);
 
   useEffect(() => {
-    fetchChildren(apiEndpoint, "published").then((data) =>
-      setColumns([
-        {
-          items: data.map((name) => ({ name, isSelected: false })),
-          selected: null,
-        },
-      ])
+    const columnPaths = columns.map((_, idx) =>
+      columns
+        .slice(0, idx + 1)
+        .map((col) => col.selected)
+        .join(".")
     );
-  }, [apiEndpoint]);
-
-  const handleItemSelected = async (itemName, columnIndex) => {
-    // Start building the path from 'published'
-    let path = "published";
-
-    // Update columns up to the current index
-    const updatedColumns = columns.slice(0, columnIndex + 1);
-    updatedColumns[columnIndex] = {
-      ...updatedColumns[columnIndex],
-      items: updatedColumns[columnIndex].items.map((item) => ({
-        ...item,
-        isSelected: item.name === itemName,
-      })),
-      selected: itemName,
-    };
-
-    // Include the selected items from previous columns in the path
-    updatedColumns.forEach((col) => {
-      if (col.selected) {
-        path += `.${col.selected}`;
-      }
+    // Fetch columns
+    const columnPromises = columnPaths.map((path, idx) =>
+      columns[idx].items.length === 0 ? fetchChildren(apiEndpoint, path) : null
+    );
+    Promise.all(columnPromises).then((responses) => {
+      setColumns((prevColumns) =>
+        prevColumns.map((column, idx) => ({
+          selected: column.selected,
+          items: responses[idx]
+            ? responses[idx].map((name) => ({
+                name,
+                isSelected: prevColumns[idx + 1]?.selected === name,
+              }))
+            : column.items,
+        }))
+      );
```

Comment:
Error handling for the `Promise.all` response is missing

Reply:
the fetch functions already handle errors internally. the promises wont throw any errors